### PR TITLE
`v1/spicepods` List API endpoints

### PR DESCRIPTION
**Pull request by Jeadie**
## Changes
- `GET /v1/spicepods`: Returns all spicepods running

Open question for the response payload for CSV (for JSON, returns `spicepod::Spicepod` ). Currently CSV rows has the following struct
```rust
pub(crate) struct SpicepodCsvRow {
        pub name: String,
        #[serde(default)]
        pub number_of_datasets: usize,
        #[serde(default)]
        pub number_of_models: usize,
        #[serde(default)]
        pub number_of_dependencies: usize,
    }
```

### CODE REVIEW

File: crates/runtime/src/http/v1.rs [hunk 1]
```diff
@@ -380,6 +387,87 @@ pub(crate) mod datasets {
...
+        format: Format,
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    #[serde(rename_all = "lowercase")]
+    pub(crate) struct SpicepodCsvRow {
+        pub name: String,
+
+        #[serde(default)]
+        pub number_of_datasets: usize,
```
Comment (lukekim) [inline crates/runtime/src/http/v1.rs]:
```suggestion
        pub datasets_count: usize,
```

File: crates/runtime/src/http/v1.rs [hunk 2]
```diff
@@ -380,6 +387,87 @@ pub(crate) mod datasets {
...
+
+        #[serde(default)]
+        pub number_of_models: usize,
```
Comment (lukekim) [inline crates/runtime/src/http/v1.rs]:
```suggestion
        pub models_count: usize,
```

File: crates/runtime/src/http/v1.rs [hunk 3]
```diff
@@ -380,6 +387,87 @@ pub(crate) mod datasets {
...
+
+        #[serde(default)]
+        pub number_of_dependencies: usize,
```
Comment (lukekim) [inline crates/runtime/src/http/v1.rs]:
```suggestion
        pub dependencies_count: usize,
```

File: bin/spice/cmd/spicepods.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import (
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spiceai/spiceai/bin/spice/pkg/api"
+	"github.com/spiceai/spiceai/bin/spice/pkg/context"
+	"github.com/spiceai/spiceai/bin/spice/pkg/util"
+)
+
+// pub dependencies: Vec<String>,
```
Comment (phillipleblanc) [inline bin/spice/cmd/spicepods.go]:
```suggestion
```

File: crates/runtime/src/http/v1.rs [hunk 5]
```diff
@@ -380,6 +387,87 @@ pub(crate) mod datasets {
...
+            }
+            Format::Csv => {
+                let resp: Vec<SpicepodCsvRow> = readable_app
+                    .spicepods
+                    .iter()
+                    .map(|spod| SpicepodCsvRow {
+                        name: spod.name.clone(),
+                        number_of_models: spod.models.len(),
+                        number_of_datasets: spod.datasets.len(),
+                        number_of_dependencies: spod.dependencies.len(),
```
Comment (lukekim) [inline crates/runtime/src/http/v1.rs]:
```suggestion
                        models_count: spod.models.len(),
                        datasets_count: spod.datasets.len(),
                        dependencies_count: spod.dependencies.len(),
```

File: crates/runtime/src/http/v1.rs (hunk 1 above)
Comment (phillipleblanc) [inline crates/runtime/src/http/v1.rs]:
```suggestion
        pub datasets_count: usize,
```

File: crates/runtime/src/http/v1.rs (hunk 2 above)
Comment (phillipleblanc) [inline crates/runtime/src/http/v1.rs]:
```suggestion
        pub models_count: usize,
```

File: crates/runtime/src/http/v1.rs (hunk 3 above)
Comment (phillipleblanc) [inline crates/runtime/src/http/v1.rs]:
```suggestion
        pub dependencies_count: usize,
```

File: crates/runtime/src/http/v1.rs [hunk 6]
```diff
@@ -380,6 +387,87 @@ pub(crate) mod datasets {
...
+        status: bool,
+
+        #[serde(default)]
+        format: Format,
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    #[serde(rename_all = "lowercase")]
+    pub(crate) struct SpicepodCsvRow {
+        pub name: String,
```
Comment (lukekim) [inline crates/runtime/src/http/v1.rs:420]: I would add `version`.

File: crates/runtime/src/http/v1.rs [hunk 7]
```diff
@@ -380,6 +387,87 @@ pub(crate) mod datasets {
...
+            Format::Json => {
+                (status::StatusCode::OK, Json(readable_app.spicepods.clone())).into_response()
+            }
+            Format::Csv => {
+                let resp: Vec<SpicepodCsvRow> = readable_app
+                    .spicepods
+                    .iter()
+                    .map(|spod| SpicepodCsvRow {
+                        name: spod.name.clone(),
+                        number_of_models: spod.models.len(),
```
Comment (phillipleblanc) [inline crates/runtime/src/http/v1.rs]:
```suggestion
                        models_count: spod.models.len(),
```

File: crates/runtime/src/http/v1.rs (hunk 5 above)
Comment (phillipleblanc) [inline crates/runtime/src/http/v1.rs]:
```suggestion
                        datasets_count: spod.datasets.len(),
                        dependencies_count: spod.dependencies.len(),
```

**Review by phillipleblanc — DISMISSED** [review]

File: crates/spicepod/src/spec.rs [hunk 8]
```diff
@@ -1,16 +1,23 @@
...
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
 #[serde(rename_all = "lowercase")]
 pub enum SpicepodVersion {
     V1Beta1,
 }
 
+impl Display for SpicepodVersion {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "{self:?}")
```
Comment (phillipleblanc) [inline crates/spicepod/src/spec.rs:17]: Is the Debug representation the most appropriate one here?

**Review by phillipleblanc — APPROVED** [review]

File: crates/spicepod/src/spec.rs (hunk 8 above)
Comment (Jeadie) [reply crates/spicepod/src/spec.rs:17]: I used this so that when SpicepodVersion.to_string(), we'd get `v1beta1` and not 0.

**Pull request merged**